hack/Dockerfile.golint

```diff
@@ -1,6 +0,0 @@
-FROM registry.hub.docker.com/library/golang:1.16
-
-RUN go get -u golang.org/x/lint/golint
-
-LABEL io.k8s.display-name="Metal3 golint test image" \
-      io.k8s.description="This image is for running tests requiring golint"
```

Comment:
IIUC this is still used in CI here: https://github.com/metal3-io/project-infra/blob/master/prow/config/config.yaml#L312-L325

Reply:
According to the https://github.com/golang/go/issues/38968 and https://github.com/golangci/golangci-lint/issues/1892, golint has been deprecated. In any case, at some point we will need to remove golint and replace it with revive. Why we are using go base image and install golint in here?, maybe we can use golangci-lint official image directly. 

Thanks to that change, we can just change golint to revive in here https://github.com/metal3-io/baremetal-operator/blob/62372c1daa01c631da46109078c9872128088f1e/.golangci.yml#L7. (I run revive in our repo and just trivial changes needed.)

cc: @honza